File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/logic/model/book/base/IPagesInfo.java
package ru.kmorozov.gbd.core.logic.model.book.base;

/**
 * Created by sbt-morozov-kv on 16.11.2016.
 */
public interface IPagesInfo {

    IPage[] getPages();

    String getMissingPagesList();

    void build();

    IPage getPageByPid(String pid);
}

File: ru.kmorozov.gbd.core.config/src/main/java/ru/kmorozov/gbd/core/logic/model/book/base/IPagesInfo.kt
package ru.kmorozov.gbd.core.logic.model.book.base

/**
 * Created by sbt-morozov-kv on 16.11.2016.
 */
interface IPagesInfo {

    val pages: Array<IPage>

    val missingPagesList: String
        get() = ""

    fun build() {}

    @Throws(PageNotFoundException::class)
    fun getPageByPid(pid: String): IPage
}
